proposals/0000-add-support-for-display-p3-colors.md

```diff
@@ -0,0 +1,209 @@
+  float green{0};
+  float blue{0};
+  float alpha{0};
+  ColorSpace colorSpace{ColorSpace::sRGB};  // Default to sRGB
+};
+```
+
+3. Update [RCTConversions](https://github.com/facebook/react-native/blob/16ad818d21773cdf25156642fae83592352ae534/packages/react-native/React/Fabric/RCTConversions.h#L37) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+4. Update [RCTTextPrimitivesConversions](https://github.com/facebook/react-native/blob/ac1cdaa71620d5bb4860237cafb108f6aeae9aef/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h#L116) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+```cpp
+auto components = facebook::react::colorComponentsFromColor(sharedColor);
+if (components.colorSpace == ColorSpace::DisplayP3) {
+  return [UIColor colorWithDisplayP3Red:components.red green:components.green blue:components.blue alpha:components.alpha];
+} else {
+  return [UIColor colorWithRed:components.red green:components.green blue:components.blue alpha:components.alpha];
+}
+```
+
+### Android Changes
+
+1. Enable the wide color gamut in [ReactActivity](https://github.com/facebook/react-native/blob/7625a502960e6b107e77542ff0d6f40fbf957322/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java#L22) if available.
+
+```java
+Window window = getWindow();
+boolean isScreenWideColorGamut = window.getDecorView().isScreenWideColorGamut()
+if (isScreenWideColorGamut) {
+  window.setColorMode(PixelFormat.COLOR_MODE_WIDE_COLOR_GAMUT)
+}
+```
+
+This is prerequisite per [Android documentation](https://developer.android.com/training/wide-color-gamut) but they note:
+
+> When wide color gamut mode is enabled, the activity's window uses more memory and GPU processing for screen composition. Before enabling wide color gamut mode, you should carefully consider if the activity truly benefits from it. For example, an activity that displays photos in fullscreen is a good candidate for wide color gamut mode, but an activity that shows small thumbnails is not.
+
+So we'll likely want to provide a way to disable this as well.
+
+This won't impact any existing 32-bit integer colors since those will continue to be in sRGB color space. To actually use DisplayP3 colors it is necessary to pack the color space and 4 color components into a 64-bit long value.
+
+2. Update [ColorUtil](https://github.com/facebook/react-native/blob/a6964b36294c3bfea09c0cdd65c5d0e3949f2dae/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ColorUtil.java#L17) and [ColorPropConverter](https://github.com/facebook/react-native/blob/781b637db4268ad7f5f3910d99ebb5203467840b/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/ColorPropConverter.java#L27) to return new color values for color space as longs.
```

Comment:
This is problematic. Changing the return type of a function will be a breaking change for everyone: if my code does
```
int c = getColor();
```
and `getColor` returns a `long`, my code will break. If I am a React Native user, which uses a library that uses this function and the library get broken for this change, I have to wait for that library to update. Which can take several weeks. Even worse, if Im depending on an unmaintained library, I might have to rewrite a huge part of my code.

I think we should brainstorm ways in which we can make this change without breaking the ecosystem. 🤔

Reply:
Right. 🤔 

Perhaps we should leave those particular functions alone but provide new functions: 
```
public static long getWideGamutColor(Object value, Context context) {}
public static boolean isWideGamutColor(Object value, Context context) {}
```

Then we can check if we have a wide gamut color and if so get the color `long` instead of the `int`. It's still unclear to me how to go about using `long` values with Android's interface components so I'm not sure if I'm missing something but it seems like it would address issues with breaking the API here.